src/esp/sensor/VisualSensor.h

```diff
@@ -18,7 +18,14 @@ class RenderTarget;
 }
 
 namespace sensor {
-
+struct VisualSensorSpec : public SensorSpec {
+  VisualSensorSpec() : SensorSpec() {
```

Comment:
Would you put the implementation to the cpp?
Also recommend using member initializer lists when giving the initial values.